File: benchmarks/c/02_reduction.c
#include <stdio.h>
#include <time.h>

#define N 1000000

double calculate_sum(double *data) {
    double total = 0.0;
    for (int i = 0; i < N; i++) {
        total += data[i];
    }
    return total;
}

int main() {
    static double data[N];
    for(int i=0; i<N; i++) data[i] = 1.0;
    
    clock_t start = clock();
    double sum = calculate_sum(data);
    clock_t end = clock();
    
    double time_spent = (double)(end - start) / CLOCKS_PER_SEC;
    printf("Sum: %.2f\n", sum);
    printf("Elapsed: %.2fs\n", time_spent);
    return 0;
}

File: benchmarks/python/02_reduction.py
import time
import random

def calculate_stats(data):
    print(f"Calculating stats for {len(data)} items...")
    
    total = 0
    max_val = float('-inf')
    min_val = float('inf')
    product = 1.0
    
    for x in data:
        total += x
        if x > max_val:
            max_val = x
        if x < min_val:
            min_val = x
        product *= (1.0 + x * 0.000001) 
        
    return total, max_val, min_val

if __name__ == "__main__":
    data = [random.random() for _ in range(1000000)]
    start = time.time()
    stats = calculate_stats(data)
    print(f"Stats: {stats}")
    print(f"Elapsed: {time.time() - start:.2f}s")
